src/Microsoft.AspNet.Mvc.Core/MvcCoreServiceCollectionExtensions.cs

```diff
@@ -153,8 +120,9 @@ private static void ConfigureDefaultServices(IServiceCollection services)
             services.AddOptions();
             services.AddRouting();
             services.AddNotifier();
-            services.Configure<RouteOptions>(
-                routeOptions => routeOptions.ConstraintMap.Add("exists", typeof(KnownRouteValueConstraint)));
+
+            services.TryAddEnumerable(
+                ServiceDescriptor.Transient<IConfigureOptions<RouteOptions>, MvcRouteOptionsSetup>());
```

Comment:
You can use `TryAddEnumerable` to make it idempotent. This is critical for *MVC Core* services.